src/platform/i386/chal/chal_prot.h

```diff
@@ -0,0 +1,73 @@
+/******************************************************************************
+Filename    : chal_prot.h
+Author      : Runyu Pan
+Date        : ?/?/2018
+Licence     : GPL v2; see COPYING for details.
+Description : The Composite chal general prototype interface.
+******************************************************************************/
+
+/* Begin Function Prototypes *************************************************/
+#ifndef CHAL_PROT
+#define CHAL_PROT
+
+/* Page table platform-dependent definitions */
+#define PGTBL_PAGEIDX_SHIFT (12)
+#define PGTBL_FRAME_BITS (32 - PGTBL_PAGEIDX_SHIFT)
+#define PGTBL_FLAG_MASK ((1 << PGTBL_PAGEIDX_SHIFT) - 1)
+#define PGTBL_FRAME_MASK (~PGTBL_FLAG_MASK)
+#define PGTBL_DEPTH 2
+#define PGTBL_ORD 10
+
+/* Page table related prototypes & structs */
+/* make it an opaque type...not to be touched */
+typedef struct pgtbl *pgtbl_t;
+
+/* identical to the capability structure */
+struct cap_pgtbl {
+	struct cap_header h;
```

Comment:
This structure looks like it would need updating for the MPU work.  Is it architecture specific?

Reply:
It is architecure-specific